apigee/apigee-devportal-kickstart-drupal
# [DRUP-728] Apigee Monetization for Kickstart

This PR adds an installation task to enable and configure Apigee Monetization for Kickstart.

During installation, if a user provides a valid Edge connection, we check if the configured organization has monetization enabled. If yes, the user can enable monetization for kickstart. 

A form to configure a store and supported currencies is also provided during installation. The form is pre-populated with values from the Apigee API.

**Note: the monetization task is optional and can be skipped. It is only available if the required modules _apigee_m10n_ and dependencies are available.**

![apigee-kickstart-DRUP-728](https://user-images.githubusercontent.com/124599/56437497-b92e4080-62f0-11e9-85de-b8c618cbdbb4.gif)

This will automatically:

1. Create an add credit product type, product variation type and order item type.
2. Configure all form displays and field types for product, variation and order item.
3. Create a default store with the Apigee organization address (configurable).
4. Import all supported currencies to the commerce store.
5. Create an add credit product for each currency.
6. Set the minimum and default top up amount from the supported currency minimum top up amount.
7. Set a product for each supported currency under the add credit config.

The only config left will be the payment gateway which we cannot configure automatically.

### How to test this PR without Monetization modules:

1. Clone the composer project `git clone git@github.com:apigee/devportal-kickstart-project-composer.git kickstart`
2. Edit `composer.json` and add the following patch: 

```
"patches": {
  "apigee/apigee_devportal_kickstart": {
    "Add monetization to kickstart": "https://patch-diff.githubusercontent.com/raw/apigee/apigee-devportal-kickstart-drupal/pull/88.diff"
  }
}
```
3. Run `composer require apigee/apigee_devportal_kickstart:dev-8.x-1.x` This will pull in the latest kickstart with this PR.

Run the installer. It should install the profile and not ask you about monetization.

### How to test this PR with Monetization modules:

1. Do step 1 to 3 above.
2. Run `composer require drupal/apigee_m10n drupal/commerce` to pull in the monetization modules and dependencies.
3. Re-run the installer. You should now see two additional tasks for setting up monetization.
